### PR TITLE
180 - Implement dataset action button component using useDatasetAction 

### PR DESCRIPTION
**Pull request by nozomione**
## Issue Number

Closing #180

- This branch was created based on [#183](https://github.com/AlexsLemonade/refinebio-web/pull/183) (will be merged first).
- This PR's commit starts at cebe9ebcf829b0025cf8f1276cde887f61ff41d7

Total files changed: 4 (new files created under `shared/DatasetActionButton/*`)

## Purpose/Implementation Notes
Based on the requirement, implemented the dataset action button component using `useDatasetAction` hook.

## Types of changes
- New feature (non-breaking change which adds functionality)

## Checklist
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/shared/DatasetActionButton/index.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/* eslint-disable react/jsx-props-no-spreading */
```
Comment (davidsmejia) [inline src/components/shared/DatasetActionButton/index.js:1]: Could you move this to the individual lines where it applies.

File: src/components/shared/DatasetActionButton/AddToDatasetButton.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Button } from 'components/shared/Button'
+
+export const AddToDatasetButton = ({
+  btnType = 'primary',
```
Comment (davidsmejia) [inline src/components/shared/DatasetActionButton/AddToDatasetButton.js:4]: let's just pass `primary` and `secondary` as props here

File: src/components/shared/DatasetActionButton/index.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import { useDatasetAction } from 'hooks/useDatasetAction'
+import { usePageRendered } from 'hooks/usePageRendered'
+import { Button } from 'components/shared/Button'
+import { AddRemainingButton } from './AddRemainingButton'
+import { AddToDatasetButton } from './AddToDatasetButton'
+import { RemoveDatasetButton } from './RemoveDatasetButton'
+
+export const DatasetActionButton = ({
+  accessionCode,
+  btnType,
```
Comment (davidsmejia) [inline src/components/shared/DatasetActionButton/index.js:12]: again lets just pass `primary` and `secondary` here

File: src/components/shared/DatasetActionButton/AddRemainingButton.js
Comment (davidsmejia) [inline src/components/shared/DatasetActionButton/AddRemainingButton.js:1]: I think for consistency, `AddRemainingDatasetButton` is more appropriate. Another option would be to rename the other components and remove `Dataset` from their names.

**Review by davidsmejia — CHANGES_REQUESTED** [review]
Looking good, just some minor changes requested.

**Comment by nozomione** [comment]
@davidsmejia based on your feedback, I've updated the codebase. Please take a look.

**Review by davidsmejia — APPROVED** [review]
LGTM

**Pull request merged**